sources/lib/api/gcp/bigquery/_udf.py

```diff
@@ -120,19 +84,77 @@ def __init__(self, inputs, outputs, name, implementation):
       outputs: a list of name/type tuples representing the schema of the output.
       name: the name of the javascript function
       implementation: a javascript function implementing the logic.
+      support_code: additional javascript code that the function can use.
+      imports: a list of GCS URLs or files containing further support code.
     Raises:
       Exception if the name is invalid.
       """
-    self._inputs = inputs
     self._outputs = outputs
     self._name = name
     self._implementation = implementation
+    self._support_code = support_code
+    self._imports = imports
+    self._code = UDF._build_js(inputs, outputs, name, implementation, support_code)
+
+  def expand_imported_code(self):
```

Comment:
It's a low cost option. BQ console does this too as it doesn't cost anything. It may be easier to debug as well (getting local console errors rather than errors back from BQ). So I'd argue for keeping it.